compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenInterfaces.td

```diff
@@ -9,6 +9,154 @@
+    case-by-case basis. The methods here are not currently complete, as we're
+    mid-refactoring. Feel free to add anything that'll be needed during the scaling
+    MMA project.
+  }];
+
+  let methods = [
+    InterfaceMethod<
+      /*desc=*/[{
+        Returns the number of non-accumulator input operands  this inner tiled
+        operation requires.
+      }],
+      /*retTy=*/"int64_t",
+      /*methodName=*/"getExpectedNumInputs",
+      /*args=*/(ins)
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        If this inner tiled operation imposes additional requirements on
+        its indexing maps other than the general ones (todo describe),
+        verify them.
+      }],
+      /*retType=*/"::mlir::LogicalResult",
+      /*methodName=*/"verifyIndexingMaps",
+      /*args=*/(ins "::llvm::ArrayRef<::mlir::AffineMap>":$maps),
+      /*methodBody=*/"",
+      /*defaultImplementation=*/[{
+        return ::mlir::success();
+      }]
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        For each input operand (includingthe accumulator, which is the final
+        operand), return the vector type of the inner tile before the operation
+        is distributed (if applicable). This should be the logical type of
+        the operands, ignoring any permutations that may be present on the
+        inner_tiled operation.
+
+        For example, a matrix multiplication that takes a MxK inner tile of A,
+        a KxN inner tile of B, and accumulates into a MxN inner tile of C
+        should return:
+        ```
+          vector<MxKxtA>, vector<KxNxtB>, vector<MxNxtC>
+        ```
+        where `tA`, `tB`, and `tC` are the element types of matrices A, B, and C,
+        respectively.
+
+        The types are placed into `result`, which has its existing contents
+        cloberred.
+      }],
+      /*retType=*/"void",
+      /*methodName=*/"getUndistributedTileTypes",
+      /*args=*/(ins "::llvm::SmallVectorImpl<::mlir::VectorType>&":$result)
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        For each input operand (including the accumulator, which is the final
+        operand), return the vector type of the inner tile after the operation
+        is distributed to threads. Vectors of this type will be stored into or
+        loaded from registers by inndividual threads.
+
+        The types are placed into `result`, which has its existing contents
+        cloberred.
+      }],
+      /*retType=*/"void",
+      /*methodName=*/"getDistributedTileTypes",
+      /*args=*/(ins "::llvm::SmallVectorImpl<::mlir::VectorType>&":$result)
+    >,
+    InterfaceMethod<
+      /*desc=*/[{
+        If this operation is to be distributed to subgroups or is required to
+        execute in a subgroup-wide way, return the size of those subgroups.
+        Otherwise, return std::nullopt.
+      }],
+      /*retType=*/"std::optional<int64_t>",
+      /*methodName=*/"getSubgroupSize",
```

Comment:
Similar comment here. I forget why we need to know this, but we should try to factor it out.